pkg/jq/jq.go

```diff
@@ -14,6 +14,12 @@ import (
 
 // Evaluate a jq expression against an input and write it to an output.
 func Evaluate(input io.Reader, output io.Writer, expr string) error {
+	return EvaluateFormatted(input, output, expr, false)
+}
+
+// Evaluate a jq expression against an input and write it to an output, either without
+// whitespace or with 2 spaces of indentation per level (pretty printed).
+func EvaluateFormatted(input io.Reader, output io.Writer, expr string, pretty_print bool) error {
```

Comment:
I would propose bringing these arguments more in line with the `pkg/jsonpretty` module from this repository: accepting `indent string` and `colorize bool` arguments and formatting the JSON accordingly:
- indent would be passed on to `enc`,
- if set, the colorize argument would cause everything to be passed through `jsonpretty`, which would write colorized, indented JSON to `output`.

BTW, Go naming conventions go against naming a variable in snake case like `pretty_print`. Instead, this would be typically named `prettyPrint`.

Reply:
Right, I had made the assumption that you wouldn't want to duplicate the colorization functionality already present in the cli packages.

If this interface adds a `colorize` flag, then jq-processed data is colorized *here* (presumably) but non-jq-transformed data is colorized there still. My PR there routes the output from this package to be colorized by the same code path as non-jq JSON data.

Reply:
I've now switched this to a `indent string, colorize bool` set of arguments, and delegate colorization to the jsonpretty package.

Note that this can _change behaviour_ when you set the indentation to the empty string but colorize to true. Where the stdlib json.Encoder outputs `[17,42]\n` in that case, jsonpretty outputs `[\n17,42\n]\n` (plus ANSI codes), putting a newline after every opening delimiter and before every closing delimiter. That might be considered a bug. At any rate, when colorize is false I don't use jsonpretty package to avoid precisely this issue.